modules/apps/client-extension/client-extension-web/src/main/java/com/liferay/client/extension/web/internal/type/deployer/CETDeployerImpl.java

```diff
@@ -117,13 +118,13 @@ protected void activate(BundleContext bundleContext) {
 	}
 
 	private String _getPortletId(CET cet) {
-		String externalReferenceCode = cet.getExternalReferenceCode();
+		ClientExtensionEntry clientExtensionEntry =
+			cet.getClientExtensionEntry();
 
 		return StringBundler.concat(
 			"com_liferay_client_extension_web_internal_portlet_",
 			"ClientExtensionEntryPortlet_",
-			externalReferenceCode.replaceAll(
-				"[^a-zA-Z0-9_]", StringPool.UNDERLINE));
+			clientExtensionEntry.getClientExtensionEntryId());
```

Comment:
:thinking: If I understood it correctly, what you really want to do is to avoid collisions between portlets with the same external reference code (ERC for short) in different companies, right?

Then, why not simply writing something like this:

```java
return StringBundler.concat(
	"com_liferay_client_extension_web_internal_portlet_",
	"ClientExtensionEntryPortlet_",
	externalReferenceCode.replaceAll(
		"[^a-zA-Z0-9_]", StringPool.UNDERLINE),
	StringPool.UNDERLINE,
	cet.getCompanyId());
```

This way, it works for Remote Apps deployed from cloud too, and we avoid breaking the `CET` abstraction.

Reply:
Got it!
Changed.
I did like this because I follow ObjectDefinition pattern.
Thanks.